### PR TITLE
adding file_name option to make_response

### PR DESCRIPTION
**Pull request by abehrens**
I wanted a convenient way to provide the file name to be presented in the content-disposition header instead of passing it in via a form.

This overrides the proxied pyexcel make* methods and adds an extra (default none) file_name option.  This could also be accomplished with a decorator, either in the client's code or within __init__.py, but I decided this was clearer.

Only the name of the file, not the extension, needs to be provided.  The extension will be added based on the file_type argument passed in.

### CODE REVIEW

File: flask_excel/__init__.py [hunk 1]
```diff
@@ -18,20 +18,37 @@ def get_file_tuple(self, field_name):
...
-    make_response_from_dict,
-    make_response_from_records,
-    make_response_from_book_dict,
-    make_response_from_a_table,
-    make_response_from_query_sets,
-    make_response_from_tables
-)
-
-__VERSION__ = '0.0.3'
+def add_file_name(response, file_name, file_type):
```
Comment (chfw) [inline flask_excel/__init__.py]: These code changes look a bit lengthy. There must be a simpler way to do this. Please allow me to re-factor it along aside with pyexcel-webio.

File: README.rst [hunk 2]
```diff
@@ -100,11 +100,15 @@ Here are some example codes::
         <p><input type=file name=file><input type=submit value=Upload>
     	</form>
         '''
-    
+
     @app.route("/download", methods=['GET'])
```
Comment (chfw) [inline README.rst]: I will remove this old 'download' code so that it looks light weight.

File: tests/test_upload_n_download_excel.py [hunk 3]
```diff
@@ -44,3 +44,10 @@ def test_download(self):
                 array = sheet.to_array()
                 assert array == self.data
 
+    def test_override_file_name(self):
```
Comment (chfw) [inline tests/test_upload_n_download_excel.py]: I like this

**Comment by chfw** [comment]
Overall, I welcome your changes. Meanwhile, I will refactor the code changes with pyexcel-webio.

File: flask_excel/__init__.py (hunk 1 above)
Comment (abehrens) [inline flask_excel/__init__.py]: I agree.  A decorator could be provided if the file name wont change based on the request.  For example:

    @download_file_name("exported_data")
    @app.route("/download")
    def download():
      return excel.make_response(....

But itd be beneficial to allow the file name arg to vary based on the request.  A decorator could be applied to the proxied pyexcel methods, too.

**Comment by abehrens** [comment]
Thanks.  I know there are a few ways to go about this but the fancier I tried to get with it via a decorator the less I liked it.  Not wedded to any one approach, as long as it's simple for the client to specify a file name, I'll be happy.

File: flask_excel/__init__.py (hunk 1 above)
Comment (chfw) [inline flask_excel/__init__.py]: I have done the code refactoring. Feel free to comment or contribute

**Comment by chfw** [comment]
This patch is released along with pyexcel-webio and others.

**Comment by abehrens** [comment]
Great! Thank you.

**Pull request merged**